src/main/java/com/internet/shop/model/User.java

```diff
@@ -19,17 +20,23 @@ public User(String name, String login, String password) {
     }
 
     public User(String name, String login, String password, Set<Role> roles) {
-        this.name = name;
-        this.login = login;
-        this.password = password;
+        this(name, login, password);
         this.roles = roles;
     }
 
+    public User(String name, String login, String password, byte[] salt, Set<Role> roles) {
+        this(name, login, password, roles);
+        this.salt = salt;
+    }
+
     public User(Long id, String name, String login, String password) {
+        this(name, login, password);
         this.id = id;
-        this.name = name;
-        this.login = login;
-        this.password = password;
+    }
+
+    public User(Long id, String name, String login, String password, byte[] salt) {
+        this(id, name, login, password);
+        this.salt = salt;
     }
```

Comment:
it's good that you have modify you constructors using this(). But do you really need such variations of them? Probably you can use setters or builder pattern WDYT?You may answer in your chat

Reply:
reduced number of constructors